src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventRecordConstructor.java

```diff
@@ -239,20 +244,32 @@ private static String getRequestUser(HookContext hookContext) {
     return requestUser == null ? hookContext.getUgi().getUserName() : requestUser;
   }
 
-  private static ExecutionMode getExecutionMode(List<ExecDriver> mrTasks, List<TezTask> tezTasks) {
-    if (tezTasks.size() > 0) {
+  private static ExecutionMode getExecutionMode(QueryPlan plan) {
+    List<ExecDriver> mrTasks = Utilities.getMRTasks(plan.getRootTasks());
+    List<TezTask> tezTasks = Utilities.getTezTasks(plan.getRootTasks());
+    List<SparkTask> sparkTasks = Utilities.getSparkTasks(plan.getRootTasks());
+    List<DDLTask> ddlTasks = TasksRetriever.getDdlTasks(plan.getRootTasks());
+
+    // Utilities methods check for null, so possibly it is nullable
+    if (plan.getRootTasks() != null && plan.getRootTasks().isEmpty()) {
+      return ExecutionMode.CLIENT_ONLY;
+    } else if (!tezTasks.isEmpty()) {
```

Comment:
`else` is not necessary if the body of the `if` above ends with `return`. Similarly for the other elses below.